components/net/http_loader.rs

```diff
@@ -437,6 +437,27 @@ fn no_referrer_when_downgrade_header(referrer_url: Url, url: Url) -> Option<Url>
     return strip_url(referrer_url, false);
 }
 
+/// https://w3c.github.io/webappsec-referrer-policy/#referrer-policy-strict-origin
+fn strict_origin(referrer_url: Url, url: Url) -> Option<Url> {
+    if referrer_url.scheme() == "https" && url.scheme() != "https" {
+        return None;
+    }
+    return strip_url(referrer_url, true);
```

Comment:
It's the last expression of the function, and the last expression in a function is the function's return value.